lib/send_grid_mailer/dev_deliverer.rb

```diff
@@ -0,0 +1,60 @@
+module SendGridMailer
+  class DevDeliverer
+    include InterceptorsHandler
+    include Logger
+    require "letter_opener"
+
+    def deliver!(sg_definition)
+      @sg_definition = sg_definition
+      execute_interceptors(@sg_definition)
+      log_definition(@sg_definition)
+      letter_opener_delivery_method.deliver!(mail)
+    end
+
+    private
+
+    def sg_api
+      @sg_api ||= Api.new(api_key)
+    end
+
+    def api_key
+      Rails.application.config.action_mailer.sendgrid_dev_settings[:api_key]
+    rescue
+      nil
+    end
+
+    def letter_opener_delivery_method
+      @letter_opener_delivery_method ||= LetterOpener::DeliveryMethod.new(location: '/tmp/mails')
+    end
+
+    def parsed_template
+      template_response = sg_api.get_template(@sg_definition)
+      template_active_version = JSON.parse(template_response.body)["versions"].find do |version|
+        version["active"] == 1
+      end
+      template_content = template_active_version["html_content"]
```

Comment:
podrías lanzar una exception si no encontrás el template. Como para tener la pista de si hemos puesto el id mal

Reply:
@ldlsegovia se supone que la exepción se lanzaría en el `api.rb` en `handle_response`. Dicho eso, probando me di cuenta que no está funcionando, ya que cuando no encuentra el recurso manda `error` en vez de `errors` y se cae el logger. 
Que crees que sea mejor, levantar una nueva excepción más custom para el caso de no encontrar el template o usar el error que existe actualmente y adaptarlo para que muestre el error que envía la api (`{"error"=>"Resource not found"}`)?

Reply:
yo creo que mejor mostrar el error de al api por si falla otra cosa, no esconder el error